src/mobu/services/github_ci/ci_manager.py

```diff
@@ -0,0 +1,337 @@
+        time.
+    http_client
+        Shared HTTP client.
+    gafaelfawr_storage
+        Gafaelfawr storage client.
+    logger
+        Global logger to use for process-wide (not monkey) logging.
+    """
+
+    shutdown_error_msg = "Mobu stopped, try re-running this check."
+
+    def __init__(
+        self,
+        users: list[User],
+        http_client: AsyncClient,
+        gafaelfawr_storage: GafaelfawrStorage,
+        logger: BoundLogger,
+    ) -> None:
+        self._users = users
+        self._gafaelfawr = gafaelfawr_storage
+        self._http_client = http_client
+        self._logger = logger.bind(ci_manager=True)
+        self._scheduler: Scheduler = Scheduler()
+        self._queue: Queue[QueueItem] = Queue()
+        self._jobs: list[Job] = []
+        self.workers: list[Worker] = []
+
+        # Used for deterministic testing
+        self.lifecycle = CiManagerLifecycle()
+
+        if not config.github_ci_app.id:
+            raise RuntimeError("MOBU_GITHUB_CI_APP_ID was not set")
+        if not config.github_ci_app.webhook_secret:
+            raise RuntimeError("MOBU_GITHUB_CI_APP_WEBHOOK_SECRET was not set")
+        if not config.github_ci_app.private_key:
+            raise RuntimeError("MOBU_GITHUB_CI_APP_PRIVATE_KEY was not set")
+        self._factory = GitHubAppClientFactory(
+            id=config.github_ci_app.id,
+            key=config.github_ci_app.private_key,
+            name="lsst-sqre/mobu CI app",
+            http_client=http_client,
+        )
+
+    async def start(self) -> None:
+        """Start the workers for the CI manager."""
+        self._logger.info("Starting CI manager...")
+        self.workers = [
+            Worker(
+                user=user,
+                queue=self._queue,
+                logger=self._logger,
+            )
+            for user in self._users
+        ]
+        self._jobs = [
+            await self._scheduler.spawn(worker.run())
+            for worker in self.workers
+        ]
+        self._logger.info("CI manager started")
+
+    async def aclose(self) -> None:
+        """Stop the workers and update GitHub CI checks for pending work.
+
+        We initially fail in-progress jobs too because we don't want any GitHub
+        check runs to be forever in-progress if mobu gets killed before they
+        finish. If they do end up finishing in time, then they'll be
+        re-concluded as successful.
+
+        We'd rather have false-negative GitHub checks than forever-in-progress
+        checks, because:
+        * There is no way to re-run in-progress checks from the GitHub UI
+        * There is no way to know for sure from the GitHub UI that a check
+          will never be concluded.
+
+        A failed check can easily be re-run by any user, so this is an
+        acceptable tradeoff, assuming false-negatives happen infrequently.
+
+        Scenarios:
+
+        ### Job completes
```

Comment:
Nit: Once we start generating Python API docs, which I usually do even for applications when they have a doc site since it can be very useful for development, it will want all of this to be reformatted as rST, not Markdown.